ui/pages/confirm-approve/confirm-approve-content/confirm-approve-content.component.test.js

```diff
@@ -281,4 +283,258 @@ describe('ConfirmApproveContent Component', () => {
+    expect(
+      queryByText('https://metamask.github.io/test-dapp/'),
+    ).toBeInTheDocument();
+    expect(getByTestId('confirm-approve-title').textContent).toStrictEqual(
+      ' Revoke permission to access and transfer all your ZenAcademy? ',
+    );
+    expect(
+      getByTestId('confirm-approve-description').textContent,
+    ).toStrictEqual(
+      ' This revokes the permission for a third party to access and transfer all your ZenAcademy without further notice. ',
+    );
+    expect(queryByText('Verify contract details')).toBeInTheDocument();
+    expect(queryByText('View full transaction details')).toBeInTheDocument();
+
+    const editButtons = getAllByText('Edit');
+
+    expect(queryByText('Transaction fee')).toBeInTheDocument();
+    expect(
+      queryByText('A fee is associated with this request.'),
+    ).toBeInTheDocument();
+    expect(queryByText(`${props.ethTransactionTotal} ETH`)).toBeInTheDocument();
+    fireEvent.click(editButtons[0]);
+    expect(props.showCustomizeGasModal).toHaveBeenCalledTimes(7);
+
+    expect(queryByText('Nonce')).toBeInTheDocument();
+    expect(queryByText('2')).toBeInTheDocument();
+    fireEvent.click(editButtons[1]);
+    expect(props.showCustomizeNonceModal).toHaveBeenCalledTimes(7);
+
+    const showViewTxDetails = getByText('View full transaction details');
+    expect(queryByText('Permission request')).not.toBeInTheDocument();
+    expect(queryByText('Approved asset:')).not.toBeInTheDocument();
+    expect(queryByText('All your')).not.toBeInTheDocument();
+    expect(queryByText('Granted to:')).not.toBeInTheDocument();
+    expect(queryByText('Data')).not.toBeInTheDocument();
+    fireEvent.click(showViewTxDetails);
+    expect(getByText('Hide full transaction details')).toBeInTheDocument();
+    expect(getByText('Permission request')).toBeInTheDocument();
+    expect(getByText('Approved asset:')).toBeInTheDocument();
+    expect(getByText('All your')).toBeInTheDocument();
+    const tokenWithSymbol = container.querySelector(
+      '.confirm-approve-content__approval-asset-title',
+    );
+    expect(tokenWithSymbol).toBeDefined();
+    expect(getByText('Granted to:')).toBeInTheDocument();
+    expect(getByText('Contract (0x9bc5baF8...fEF4)')).toBeInTheDocument();
+    expect(getByText('Data')).toBeInTheDocument();
+    expect(getByText('Function: SetApprovalForAll')).toBeInTheDocument();
+    expect(getByText('Parameters: false')).toBeInTheDocument();
+    expect(
+      getByText(
+        '0x095ea7b30000000000000000000000009bc5baf874d2da8d216ae9f137804184ee5afef40000000000000000000000000000000000000000000000000000000000011170',
+      ),
+    ).toBeInTheDocument();
+  });
+
+  it('should render Confirm approve page correctly and display fallback copy for when we are not able to retrieve a erc721 or erc1155 name in the setApprovalForAll screen and when revoking a setApprovalForAll allowance', () => {
+    const { queryByText, getByText, getAllByText, getByTestId, container } =
```

Comment:
Snapshot test for specific part of the page might be better and more readable for this kind of test.

Reply:
@jpuri Done [here](https://github.com/MetaMask/metamask-extension/pull/17992/commits/d7e12d4155ab6ed63a4b87f71350d9d801457c99#diff-49a399717cd27ce0bb0c752633bd4e684dd9ae6469f26f0a1dcaf607ffdc6d76).